Feature/Improve release process

carpawell opened the pull request:
Release binaries are not built in the clean reproducible environment. It would be safer to build binaries in container, the same way we do for docker images.

alexvanin reviewed (commented):
Works great. The only thing I am not sure: do we really need to build images in `make prepare-release`? Taking into account that image deploy will be automated by @sami-nspcc soon, I suppose we can leave only binary \ tar generation routine.

carpawell commented:
@alexvanin, sure

deleted all the thing about docker

realloc commented:
I'd suggest doing another thing instead:
- Add clean build in docker environment, the same way it's done for [neofs-spec](https://github.com/nspcc-dev/neofs-spec/blob/master/Makefile#L133)
- Leave release binaries creation to @sami-nspcc
- Let @sami-nspcc build and attach binaries for each new release page she will notice

carpawell commented:
@realloc I made a mistake: it is golang ALPINE image that does not contain `make` util. So I updated Makefile with `docker/%` target since, I think, it is OK to build binaries not with `alpine` image.

@alexvanin, @cthulhu-rider, please, take a look at https://github.com/nspcc-dev/neofs-node/pull/785/commits/13a5952723a776f92b1de460e35794b21cf9d6ed: I added updating go instruction and there is one more place that contains go version

realloc commented on docs/release-instruction.md:
```diff
@@ -26,33 +28,36 @@ Use `vX.Y.Z` tag for releases and `vX.Y.Z-rc.N` for release candidates
-    $ docker push nspccdev/neofs-storage:0.12.0
-    $ docker push nspccdev/neofs-ir:0.12.0
-    $ docker push nspccdev/neofs-cli:0.12.0
+    $ make images && make image-storage-testnet
```
Can we add `image-storage-testnet` into the `images` target deps?

carpawell replied:
sure

realloc commented on Makefile:
```diff
@@ -9,14 +9,21 @@ DEBUG ?= false
 # List of binaries to build.
 CMDS = $(notdir $(basename $(wildcard cmd/*)))
 BINS = $(addprefix $(BIN)/, $(CMDS))
+RELEASE_BINS = $(foreach ARCH,$(ARCHS),$(addsuffix -$(ARCH), $(addprefix $(RELEASE)/, $(CMDS))))
```
Why it's a separate list? Shuldn't the regular `bin` and release `bin` be the same?

carpawell replied:
because i think of `make all` as "I want to have neofs binaries in my local env for using it/testing it", so it shouldn't be with arch suffixes and I don't want to have all arch variations to be build

on the other hand, `make prepare-release` is something that shouldn't be used by devs every day, so I think it should be used separately

realloc commented on Makefile:
```diff
@@ -37,6 +44,28 @@ $(DIRS):
+# Prepare binaries and archives for release
+prepare-$(RELEASE): $(RELEASE) $(RELEASE_BINS)
+
+$(RELEASE)/%: dep
```
This looks the same way as regular `dep`, apart from arch support. I suggest adding different arch support as a separate change when we need it. Currently, we support only amd64.

carpawell replied:
why can't it be like already done? do not see any problems with that

at least it is different because of different targets that do not intersect(`BINS` and `RELEASE_BINS`)

carpawell commented:
@realloc, @alexvanin, simplified release target
